nuitka/nodes/ListNodes.py

```diff
@@ -0,0 +1,317 @@
+#     Copyright 2022, Kay Hayen, mailto:kay.hayen@gmail.com
+#
+#     Part of "Nuitka", an optimizing Python compiler that is compatible and
+#     integrates with CPython, but also works on its own.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+""" Nodes that build and operate on lists."""
+
+
+
+from abc import abstractmethod
+
+from .ConstantRefNodes import makeConstantRefNode
+from .ExpressionBases import (
+    ExpressionChildHavingBase,
+    ExpressionChildrenHavingBase,
+)
+from .ExpressionShapeMixins import (
+    ExpressionBoolShapeExactMixin,
+    ExpressionBytesShapeExactMixin,
+    ExpressionIntShapeExactMixin,
+    ExpressionListShapeExactMixin,
+    ExpressionStrOrUnicodeExactMixin,
+    ExpressionStrShapeExactMixin,
+    ExpressionTupleShapeExactMixin,
+)
+from .NodeBases import SideEffectsFromChildrenMixin
+from .NodeMetaClasses import NodeCheckMetaClass
+
+
+class ExpressionListOperationAppend(
+    ExpressionListShapeExactMixin, ExpressionChildHavingBase
+):
+    """This operation represents l.append(object)."""
+
+    kind = "EXPRESSION_LIST_OPERATION_APPEND"
+
+    name_children = ('list_arg', 'item')
+
+    def __init__(self, list_arg, item, source_ref):
+        assert list_arg is not None
+        assert item is not None
+
+
+        ExpressionChildHavingBase.__init__(
+            self,
+            value={'list_arg': list_arg, 'item': item},
+            source_ref=source_ref,
+        )
+
+    def computeExpression(self, trace_collection):
+        list_arg = self.subnode_list_arg
+        item = self.subnode_item
```

Comment:
You should also do this:

```
            # They take over the variable content, exposing it to currently untraced usages.
            item.onContentEscapes(trace_collection)
```

Through being appended to the list, we loose track of values, and should take into account, that other places get access to it.